### PR TITLE
Present which files have conflicts to the user during upgrades

### PR DESCRIPTION
**Pull request by aclevername**
### Description

New UX when a conflict occurs:
```
$ pctl upgrade --git-repository foo/bar weaveworks-nginx/ v0.1.1
upgrading profile "pctl-profile" from version "v0.1.0" to "v0.1.1"
upgrade succeeded but merge conflict have occured, please resolve manually. Files containing conflicts:
- weaveworks-nginx/artifacts/nginx-deployment/nginx/deployment/deployment.yaml
```

### Checklist
- [ ] Added tests that cover your change
- [ ] Added/modified documentation as required (such as the `README.md`)
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] Added a `kind` label to the PR (e.g. `kind/feature`)

### CODE REVIEW

File: pkg/upgrade/upgrade.go [hunk 1]
```diff
@@ -148,5 +144,15 @@ func Upgrade(cfg UpgradeConfig) error {
 		return fmt.Errorf("failed to copy upgraded installation into installation directory: %w", err)
 	}
 
+	if len(mergeConflictFiles) > 0 {
+		msg := "upgrade succeeded but merge conflict have occured, please resolve manually. Files containing conflicts:\n"
```
Comment (Callisto13) [inline pkg/upgrade/upgrade.go:148]:
```suggestion
		msg := "upgrade succeeded but merge conflicts have occured, please resolve manually. Files containing conflicts:\n"
```

File: pkg/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -144,6 +144,18 @@ var _ = Describe("Upgrade", func() {
...
 	})
 
+	When("merge conflicts occur", func() {
+		BeforeEach(func() {
+			fakeRepoManager.MergeBranchesReturns([]string{"foo/bar"}, nil)
+		})
+
+		It("returns a list of files that contain conflicts", func() {
+			err := upgrade.Upgrade(cfg)
+			expectedErrMsg := fmt.Sprintf("upgrade succeeded but merge conflict have occured, please resolve manually. Files containing conflicts:\n- %s", filepath.Join(profileDir, "foo/bar"))
```
Comment (Callisto13) [inline pkg/upgrade/upgrade_test.go:154]:
```suggestion
			expectedErrMsg := fmt.Sprintf("upgrade succeeded but merge conflicts have occurred, please resolve manually. Files containing conflicts:\n- %s", filepath.Join(profileDir, "foo/bar"))
```

File: tests/integration/e2e_test.go [hunk 3]
```diff
@@ -190,4 +190,131 @@ status: {}
...
+				fmt.Sprintf("%s/%s", namespace, gitRepoName),
+				profileDir,
+				"v0.1.1")
+			cmd.Dir = temp
+			output, err = cmd.CombinedOutput()
+			Expect(err).To(HaveOccurred())
+			Expect(strings.Split(string(output), "\n")).To(
+				ConsistOf(
+					`upgrading profile "pctl-profile" from version "v0.1.0" to "v0.1.1"`,
+					"upgrade succeeded but merge conflict have occured, please resolve manually. Files containing conflicts:",
```
Comment (Callisto13) [inline tests/integration/e2e_test.go:282]:
```suggestion
					"upgrade succeeded but merge conflicts have occurred, please resolve manually. Files containing conflicts:",
```

**Review by Callisto13 — APPROVED** [review]
LGTM, just two small typos and because I am AWESOME I have changed them everywhere for you. yw, I will take my reward in cash please

**Comment by aclevername** [comment]
> LGTM, just two small typos and because I am AWESOME I have changed them everywhere for you. yw, I will take my reward in cash please

money is in the post, ty :bow:

**Pull request merged**